scripts/tier_analyzer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+import argparse
+import re
+import subprocess
+import sys
+
+
+def check_virtualenv():
+    """Check if we are in virtualenv and if not, exit."""
+    if not hasattr(sys, 'real_prefix'):
+        print('You must activate CFME virtualenv in oder to run this script.')
+        sys.exit(1)
+
+
+def get_pytest_output(args, use_tier_marker=False):
+    """Get string output of pytest command."""
+    print('Getting pytest output {} tier marker.'.format('with' if use_tier_marker else 'without'))
+    pytest_args = ['pytest',
+        args.test_path,
+        '--collect-only',
+        '--long-running',
+        '--use-provider',
+        args.provider,
+        '-m']
+    if use_tier_marker:
+        pytest_args.append(args.tier_marker)
+    else:
+        pytest_args.append('uses_testgen')
+
+    return subprocess.check_output(pytest_args)
+
+
+def parse_pytest_output(output):
```

Comment:
**REQUIRED:** This is not generically parsing pytest output.   It is parsing out the parameterized test cases.   The name should change to reflect that, such as parse_testcases_from_pytest_output, or get_testcases_from_pytest_output.

Just something to think about, you could even change to get_pytest_testcases_for_provider(), and then call the function that runs pytest and gets the output from here.   That way your main function just calls this function.   It's fine the way it is, but it would make the main function show less knowledge about the details and only at a high level what it is doing, which is getting the test cases for a particular provider with and without filtering on a tier.

Reply:
And you could also argue that your current way is simpler.   I'm just offering a different perspective of simple.

Reply:
@jamesooden I understand your argument and I don't even want to make the case that my current implementation is better. I just think that it is not very defensible from the ROI point of view to put so much effort into design of such a narrow utility script. That's why I'd like to go with what I have. Not because it is necessarily better, but because it's not really incorrect and gets the job done.

Reply:
I mainly wanted to pass down a design principle of keeping implementation details hidden and providing abstractions that keep the user of the abstraction, and in this case the reader of the code focused on the problem to be solved rather than the details of how it is being solved.    So mission accomplished.